### ISSUE TITLE
Drawer A11y issues

### ISSUE THREAD

**Issue by mbmjertan** [infinum/eightshift-frontend-libs#534]
### Describe your bug

Currently, eightshift-frontend-libs provides a drawer component which uses a `transform: translate3d(100%, 0, 0);` rule (with `display: block;`) to hide itself and animate opening/closing. It also uses a page overlay component which renders a `.page-overlay` div to darken the rest of the page (and allow the drawer to be closed by clicking on the page itself)

However, this approach introduces some A11y issues:

- [ ] As the drawer has `display: block`, its elements are part of the tab order, even in cases where a drawer trigger isn't rendered (such as on desktop) or drawer is closed - this creates confusion for keyboard users
- [ ] Clickable elements must be focusable and should have interactive semantics - https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard

Generally speaking, keyboard control could be improved here (e.g., supporting `esc` to close the drawer)

### Steps to Reproduce

1. Set up a menu on the default eightshift-boilerplate.

### Expected behavior

Drawer elements aren't part of the tab order when the drawer is closed. Drawer is fully keyboard accessible. 

### Screenshots, screen recording, code snippet

<img width="1037" alt="Screenshot 2022-01-10 at 10 14 42" src="https://user-images.githubusercontent.com/1742806/148748312-506f8f1e-155a-47ba-a336-3d2c2035ef35.png">
<img width="703" alt="Screenshot 2022-01-10 at 10 15 14" src="https://user-images.githubusercontent.com/1742806/148748321-2c0df3d7-2539-40eb-a697-85844c7a399a.png">

### Please confirm that you have searched existing issues in this repo.

Yes

### Please confirm that you have searched in our documentation and couldn't find the answer.

Yes

### Please confirm that your bug occurs with all plugins uninstalled and with the default WordPress theme active.

No
